exporter/exporter_test.go

```diff
@@ -54,19 +54,44 @@ func getSecure(t *testing.T, url string) (*http.Response, error) {
 	return httpClient.Get(url)
 }
 
-func checkExporter(t *testing.T, addr string, secure bool) error {
+func buildExporterURL(user, pass, addr string, secure bool) string {
+
+	proto := "http"
+	if secure {
+		proto = "https"
+	}
+
+	if user != "" {
+		return fmt.Sprintf("%s://%s:%s@%s/metrics", proto, user, pass, addr)
+	}
+
+	return fmt.Sprintf("%s://%s/metrics", proto, addr)
+}
+
+func checkExporterFull(t *testing.T, user, pass, addr string, secure bool, expectedRc int) error {
 	var resp *http.Response
 	var err error
+	url := buildExporterURL(user, pass, addr, secure)
+
 	if secure {
-		resp, err = getSecure(t, fmt.Sprintf("https://%s/metrics", addr))
+		resp, err = getSecure(t, url)
 	} else {
-		resp, err = http.Get(fmt.Sprintf("http://%s/metrics", addr))
+		resp, err = http.Get(url)
 	}
 	if err != nil {
 		return err
 	}
-	if resp.StatusCode != 200 {
-		return fmt.Errorf("expected a 200 response, got %d", resp.StatusCode)
+	// Avoid EOF errors in Travis
+	resp.Close = true
+
+	rc := resp.StatusCode
+	if rc != expectedRc {
+		return fmt.Errorf("expected a %d response, got %d", expectedRc, rc)
+	}
+
+	// bail on auth error, etc.
+	if rc != 200 {
+		return nil
```

Comment:
So we test above for the expectedRc, but if we are here and it is not a `200`, then we return no error. The close of `resp.Body` was missing. Having the `defer` above as mentioned will solve that.

Reply:
Ah, yep.  Thanks.